cmd/goal/clerk.go

```diff
@@ -1056,6 +1069,17 @@ var compileCmd = &cobra.Command{
 					reportErrorf("%s: %s", outname, err)
 				}
 			}
+			if writeSourceMap {
+				mapname := fname + ".map"
```

Comment:
@algochoi small question/request, was there a specific reason you chose `fname + ".map"` for the map file name? With this the source map is given the name of the input file, which I think is less intuitive/accurate.

In JS it's common practice for the map file to be named after the produced output file, since that's the starting point from which you can go "backwards" with the map, so I was expecting `outname + ".map"` here (idk what should happen when the output is stdout, maybe an error?).

Reply:
> With this the source map is given the name of the input file, which I think is less intuitive/accurate.

That is a fair point, and I can see the confusion. I can open up a PR to fix this. 

> idk what should happen when the output is stdout, maybe an error?

This what I initially thought, but after playing around with the commands, I think the source map should always be output to the file because the `no-out` flag says `don't write contract program binary`, and a source map isn't necessarily the binary. 

But alternatively, we could also output the source map to stdout if the no output flag is raised. That sounds fine too, depending on how you interpret the `no-out` flag.

Reply:
> That is a fair point, and I can see the confusion. I can open up a PR to fix this.

That would be great, thanks!

> I think the source map should always be output to the file because the `no-out` flag says `don't write contract program binary`, and a source map isn't necessarily the binary.

That's definitely a valid interpretation, but what do you name the source map file then? I think `<input>.tok.map` would work.

Though to be fair, I think it's quite rare to use the `no-out` flag, so failing in this case would not be that bad in my opinion.